### PR TITLE
Bugfix: remove old driver syntax

### PR DESCRIPTION
**Pull request by thomasgibson**
# Description

A particular code path in the multirate solver driver was never tested, so this particular problem wasn't caught. Basically some stale code in the driver setup wasn't updated. This PR fixes the bug and updates a unit test.

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [x] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [x] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: src/Driver/SolverTypes/MultirateSolverType.jl [hunk 1]
```diff
@@ -333,20 +333,17 @@ function solversetup(
...
-        diffusion_direction = diffusion_direction,
+    remainder_kwargs = (
+        numerical_flux_first_order = (
+            ode_solver.discrete_splitting ?
+                    (
+                dg.numerical_flux_first_order,
+                (dg.numerical_flux_first_order,),
+            ) :
+                    dg.numerical_flux_first_order
+        ),
```
Comment (kpamnany) [inline src/Driver/SolverTypes/MultirateSolverType.jl:344]: Maybe just make this an `if` block? This is hard to read.

**Comment by thomasgibson** [comment]
bors retry

**Pull request merged**